extensions/infinispan-client/runtime/src/main/java/io/quarkus/infinispan/client/runtime/InfinispanClientRuntimeConfig.java

```diff
@@ -21,10 +21,31 @@
 @ConfigRoot(name = "infinispan-client", phase = ConfigPhase.RUN_TIME)
 public class InfinispanClientRuntimeConfig {
 
+    // @formatter:off
+    /**
+     * Sets the URI of the running Infinispan server to connect to. hotrod://localhost:11222@admin:password
+     * If provided {@link #hosts}, {@link #username} and {@link #password} will be ignored.
+     */
+    // @formatter:on
+    @ConfigItem
+    public Optional<String> uri;
+
+    // @formatter:off
+    /**
+     * Sets the host name/port to connect to. Each one is separated by a semicolon (eg. host1:11222;host2:11222).
+     */
+    // @formatter:on
+    @ConfigItem
+    public Optional<String> hosts;
+
+    // @formatter:off
     /**
      * Sets the host name/port to connect to. Each one is separated by a semicolon (eg. host1:11222;host2:11222).
+     * @deprecated {@link #hosts} should be used to configure the list or uri for an uri connection string.
      */
+    // @formatter:on
     @ConfigItem
+    @Deprecated
```

Comment:
We should mention which property replaces it

Reply:
I'm mentioning that in the docs above.

Reply:
Sure, but anyone reading the Javadoc needs to look at another source of information to find it - that is not a great user experience

Reply:
I'm adding the tags in the javadocs too...

Reply:
Great, I see